### PR TITLE
Fixes #24 and Fixes #19 - big code cleanup

### LINKED ISSUE #19: AUV - too many campains. can't filter
The auv viewer becomes clogged up with the quantity of repeated dives in similar places. It starts to be almost impossible to select a dive a user is interested in. If a user selects a dive from the dropdown menu, and click on one track, it's impossible to find the track he really is interested in. I can't.

## solution
Maybe some basic CQL filtering could fix this. A user could tick a box somewhere on the main page to display only the dive/campaign he is interested in.

Not many hours of work have been allocated to the viewer in the last few years. This feature has been asked for a long time, maybe since the beginning. Looking at the commit log https://github.com/aodn/auv/commits/master really shows that most of the commits are just basic maintenance.

### LINKED ISSUE #24: visualisation bugs with viewer
As reported by Neville Barrett who actively uses the viewer
(edit : he confirmed me that the following glintches are only recent)
Example around this location 
```
Lon: 151.26242
Lat: -34.00703
```

Many visualisation bugs exist at different zoom level

** Medium zoom level
The two dives (visible with a yellow auv icon), aren't always visible. The one on the right disappears at a certain zoom level.

and clicking on the icons don't always give results

** track zoom level
The track is not displayed all the time as a track anymore (? maybe due to the last openlayer update ?). Random dots appear, and if the user keep on zooming, he will eventually see the proper track

another good example is with the Tasmania 2015 campaign at 
```
Lon: 146.28512
Lat: -43.74408
```

### PR DESCRIPTION
**Pull request by pmbohm**

### CODE REVIEW

**Comment by pmbohm** [comment]
This is a big cleanup of code that hasn't been touched for years. The aim here was to understand the code again, without doing a rewrite as much as that may have been a good idea. This app is expected to be replace in the not too distant future and these changes should be seen as doing enough for that aim.

Note: If testing this code, note that it relies on https://github.com/aodn/geoserver-config/issues/271 to be fixed. Clicking on an AUV icon in the map will not work. Also many sites will be empty, showing no tracks or image locations

File: grails-app/views/finder/_auv_functions_js.gsp [hunk 1]
```diff
@@ -320,12 +300,12 @@
         bbox = bb[0] + "," + bb[1] + "%20" + bb[2] + "," + bb[3];
 
         var filter = encodeURIComponent("<ogc:Filter xmlns:ogc=\"http://ogc.org\" xmlns:gml=\"http://www.opengis.net/gml\"><ogc:BBOX>"
-                + "<ogc:PropertyName>geom</ogc:PropertyName>"
-                + "<gml:Box srsName=\"http://www.opengis.net/gml/srs/epsg.xml\">"
-                + "<gml:coordinates>")
+                        + "<ogc:PropertyName>geom</ogc:PropertyName>"
+                        + "<gml:Box srsName=\"http://www.opengis.net/gml/srs/epsg.xml\">"
+                        + "<gml:coordinates>")
```
Comment (danfruehauf) [inline grails-app/views/finder/_auv_functions_js.gsp]: You messed here the indent...

File: application.properties [hunk 2]
```diff
@@ -3,6 +3,6 @@
 app.grails.version=1.3.7
 app.name=auv
 app.servlet.version=2.4
-app.version=0.1.4
+app.version=0.1.5
```
Comment (danfruehauf) [inline application.properties]: No need to do that, Jenkins will do it for you.

**Comment by danfruehauf** [comment]
I think you squashed this with a commit that's not yours. Notice the author of the commit is jenkins.

File: grails-app/views/finder/_auv_functions_js.gsp [hunk 3]
```diff
@@ -497,68 +469,55 @@
...
-        var html_content = "";
-
-        if (tmp_response.match(/<\/body>/m)) {
-            html_content = tmp_response.match(/(.|\s)*?<body[^>]*>((.|\s)*?)<\/body>(.|\s)*?/m);
-            if (html_content) {
-                //trimmed_content= html_content[2].replace(/(\n|\r|\s)/mg, ''); // replace all whitespace
-                html_content = html_content[2].replace(/^\s+|\s+$/g, '');  // trim
-            }
-        }
+        var html_content = getTrimBodyContent(tmp_response);
```
Comment (danfruehauf) [inline grails-app/views/finder/_auv_functions_js.gsp]: Perhaps modify here too to `htmlContent` and `tmpResponse`

File: grails-app/views/finder/_auv_functions_js.gsp [hunk 4]
```diff
@@ -767,15 +588,34 @@
...
+    function setSiteCodeCql(site_code) {
+
+        var wmsLayers = map.getLayersByClass("OpenLayers.Layer.WMS");
+        for (var key in wmsLayers) {
+            var layer = map.getLayer(map.layers[key].id);
+            var layerName = layer.params.LAYERS;
+            if (layerName == fqLayerNameTracks || layerName == fqLayerNameImages) {
+                if (site_code) {
+                    layer.mergeNewParams({
+                        CQL_FILTER: "site_code like '" + site_code + "'"
```
Comment (danfruehauf) [inline grails-app/views/finder/_auv_functions_js.gsp]: I guess this is the essence of this PR, correct me if I'm wrong...

File: grails-app/views/finder/_auv_functions_js.gsp (hunk 3 above)
Comment (pmbohm) [inline grails-app/views/finder/_auv_functions_js.gsp]: Its not only here that camel case isn't used. I don't want to go through the code changing everything to camel case. 
"This app is expected to be replaced in the not too distant future"

Comment (danfruehauf) [inline grails-app/views/finder/_auv_functions_js.gsp]: Don't rely on that.

**Comment by pmbohm** [comment]
@danfruehauf fresh commit now pushed

**Pull request merged**